src/praisonai-deploy/tests/test_cli.py

```
@@ -0,0 +1,390 @@
+def test_build_deploy_config_from_args_api_yaml_wins_when_cli_unset():
+    """API shortcut must honor deploy.api YAML when CLI flags are unset (None)."""
+    from types import SimpleNamespace
+    import yaml
+    from praisonai_deploy.cli.features.deploy import _build_deploy_config_from_args
+    from praisonai_deploy.models import DeployType
+
+    with tempfile.NamedTemporaryFile("w", suffix=".yaml", delete=False) as f:
+        yaml.safe_dump({
+            "deploy": {
+                "type": "api",
+                "api": {"host": "0.0.0.0", "port": 9000, "workers": 4},
```

Ruff (0.16.0)

[error] Possible binding to all interfaces

(S104)